### PR TITLE
Added websocket support and missing Getorderbook call

### PR DESCRIPTION
**Pull request by saniales**

### CODE REVIEW

**Comment by bitbandi** [comment]
I fixed the file indents in my repo, could you rebase the commits to current master?
and split the websocket.go into a subdirectory?

File: websocket.go [hunk 1]
```diff
@@ -0,0 +1,207 @@
...
+	}, nil
+}
+
+// Close closes the Websocket connected to the hitbtc api.
+func (c *WSClient) Close() {
+	c.conn.Close()
+}
+
+// WSGetCurrencyRequest is get currency request type on websocket
+type WSGetCurrencyRequest struct {
```
Comment (bitbandi) [inline websocket.go:173]: websocket/WSGetCurrency.go

File: websocket.go [hunk 2]
```diff
@@ -0,0 +1,207 @@
...
+	Currency string `json:"currency,required"`
+}
+
+// WSGetCurrencyResponse is get currency response type on websocket
+type WSGetCurrencyResponse struct {
```
Comment (bitbandi) [inline websocket.go:178]: websocket/WSGetCurrency.go

File: websocket.go [hunk 3]
```diff
@@ -0,0 +1,207 @@
...
+
+	err := c.conn.Call(context.Background(), "getCurrency", request, &response)
+	if err != nil {
+		return nil, errors.Annotate(err, "Hitbtc GetCurrency")
+	}
+	return &response, nil
+}
+
+// WSGetSymbolsRequest is get symbols request type on websocket
+type WSGetSymbolsRequest struct {
```
Comment (bitbandi) [inline websocket.go]: websocket/WSGetSymbols.go

File: websocket.go [hunk 4]
```diff
@@ -0,0 +1,207 @@
...
+	Symbol string `json:"symbol,required"`
+}
+
+// WSGetSymbolsResponse  is get symbols response type on websocket
+type WSGetSymbolsResponse struct {
```
Comment (bitbandi) [inline websocket.go]: websocket/WSGetSymbols.go

File: websocket.go [hunk 5]
```diff
@@ -0,0 +1,207 @@
...
+	QuoteCurrency        string `json:"quoteCurrency,required"`
+	QuantityIncrement    string `json:"quantityIncrement,required"`
+	TickSize             string `json:"tickSize,required"`
+	TakeLiquidityRate    string `json:"takeLiquidityRate,required"`
+	ProvideLiquidityRate string `json:"provideLiquidityRate,required"`
+	FeeCurrency          string `json:"feeCurrency,required"`
+}
+
+// WSSubscribeTickerRequest is get symbols request type on websocket
+type WSSubscribeTickerRequest struct {
```
Comment (bitbandi) [inline websocket.go]: websocket/WSSubscribeTicker.go

File: websocket.go [hunk 6]
```diff
@@ -0,0 +1,207 @@
...
+	Symbol string `json:"symbol,required"`
+}
+
+// WSNotificationTickerResponse is notification response type on websocket
+type WSNotificationTickerResponse struct {
```
Comment (bitbandi) [inline websocket.go:270]: websocket/WSNotificationTicker.go

File: websocket.go [hunk 7]
```diff
@@ -0,0 +1,207 @@
...
+}
+
+// WSSubtypeTrade is element of market trade type
+type WSSubtypeTrade struct {
+	Price string `json:"price,required"`
+	Size  string `json:"size,required"`
+}
+
+// WSNotificationOrderbookSnapshot is notification response type to orderbook snapshot on websocket
+type WSNotificationOrderbookSnapshot struct {
```
Comment (bitbandi) [inline websocket.go:374]: websocket/WSNotificationOrderbook.go

File: websocket.go [hunk 8]
```diff
@@ -0,0 +1,207 @@
...
+	Ask      []WSSubtypeTrade `json:"ask,required"`
+	Bid      []WSSubtypeTrade `json:"bid,required"`
+	Symbol   string           `json:"symbol,required"`
+	Sequence int64            `json:"sequence,required"`
+}
+
+// WSNotificationOrderbookUpdate is notification response type to orderbook snapshot on websocket
+type WSNotificationOrderbookUpdate struct {
```
Comment (bitbandi) [inline websocket.go:382]: websocket/WSNotificationOrderbook.go

File: websocket.go [hunk 9]
```diff
@@ -0,0 +1,207 @@
...
+type WSTrades struct {
+	ID        int    `json:"id,required"`
+	Price     string `json:"price,required"`
+	Quantity  string `json:"quantity"`
+	Side      string `json:"side,required"`
+	Timestamp string `json:"timestamp,required"`
+}
+
+// WSNotificationTradesSnapshot is notification response type to trades on websocket
+type WSNotificationTradesSnapshot struct {
```
Comment (bitbandi) [inline websocket.go]: websocket/WSNotificationTrades.go

File: websocket.go [hunk 10]
```diff
@@ -0,0 +1,207 @@
...
+	Data []WSTrades `json:"data,required"`
+}
+
+// WSNotificationTradesUpdate is notification response type to trades on websocket
+type WSNotificationTradesUpdate struct {
```
Comment (bitbandi) [inline websocket.go]: websocket/WSNotificationTrades.go

File: websocket.go [hunk 11]
```diff
@@ -0,0 +1,207 @@
...
+	Data WSTrades `json:"data,required"`
+}
+
+// WSGetTradesRequest is get trades request type on websocket
+type WSGetTradesRequest struct {
```
Comment (bitbandi) [inline websocket.go]: websocket/WSGetTrades.go

File: websocket.go [hunk 12]
```diff
@@ -0,0 +1,207 @@
...
+	Limit  int       `json:"limit,required"`
+	Sort   string    `json:"sort,required"`
+	By     string    `json:"by,required"`
+	From   time.Time `json:"from"`
+	Till   time.Time `json:"till"`
+	Offset string    `json:"offset"`
+}
+
+// WSGetTradesResponse  is get symbols response type on websocket
+type WSGetTradesResponse struct {
```
Comment (bitbandi) [inline websocket.go]: websocket/WSGetTrades.go

File: websocket.go [hunk 13]
```diff
@@ -0,0 +1,207 @@
...
+	Open        string    `json:"open,required"`
+	Close       string    `json:"close,required"`
+	Min         string    `json:"min,required"`
+	Max         string    `json:"max,required"`
+	Volume      string    `json:"volume,required"`      // Total trading amount within 24 hours in base currency
+	VolumeQuote string    `json:"volumeQuote,required"` // Total trading amount within 24 hours in quote currency
+}
+
+// WSSubscribeCandlesNotificationSnapshot is subscribe response type to candles on websocket
+type WSSubscribeCandlesNotificationSnapshot struct {
```
Comment (bitbandi) [inline websocket.go]: websocket/WSCandles.go

File: websocket.go [hunk 14]
```diff
@@ -0,0 +1,207 @@
...
+	Data   []WSCandles `json:"data,required"`
+	Symbol string      `json:"symbol,required"`
+	Period string      `json:"period,required"`
+}
+
+// WSSubscribeCandlesNotificationUpdate is subscribe response type to candles on websocket
+type WSSubscribeCandlesNotificationUpdate struct {
```
Comment (bitbandi) [inline websocket.go]: websocket/WSCandles.go

File: websocket.go [hunk 15]
```diff
@@ -0,0 +1,207 @@
...
+	Data []WSTrades `json:"data,required"`
+}
+
+// WSSubscribeCandlesRequest is subscribe request type to candles on websocket
+type WSSubscribeCandlesRequest struct {
+	Symbol string `json:"symbol,required"`
+	Period string `json:"period,required"`
+}
+
+// WSCandles is item for WSCandles
```
Comment (bitbandi) [inline websocket.go:450]: websocket/WSCandles.go

File: websocket.go [hunk 16]
```diff
@@ -0,0 +1,207 @@
...
+	Offset string    `json:"offset"`
+}
+
+// WSGetTradesResponse  is get symbols response type on websocket
+type WSGetTradesResponse struct {
+	Data []WSTrades `json:"data,required"`
+}
+
+// WSSubscribeCandlesRequest is subscribe request type to candles on websocket
+type WSSubscribeCandlesRequest struct {
```
Comment (bitbandi) [inline websocket.go]: websocket/WSCandles.go

**Comment by saniales** [comment]
I think its better to keep same package, due to go coding style rules
https://golang.org/doc/code.html

**Comment by saniales** [comment]
regarding the rebase I will do it ASAP

**Comment by saniales** [comment]
@bitbandi I rebased, but i strongly ask you to keep websocket content code in websocket.go and avoiding to split as it goes against go official guidelines. One file per concept, one package per area.
Regarding this we should coordinate together to refactor the code in another PR and find a common point of View. Let's do this, but for now just merge it :)

**Comment by saniales** [comment]
Additional reason why keeping same package seems best to me is practical usage
``` go
wsClient, err := hitbtc.NewWSClient(...)
VS
wsClient, err := websocket.NewClient(...)
```

in this way in a multi-exchange  context, like arbitrage bots, you can clearly see from package name that the client comes for hitbtc, and also avoids multiple `websocket` package naming changes.

**Comment by saniales** [comment]
@bitbandi would you merge this, please?

**Pull request closed without merging**